### ISSUE TITLE
Duplicate middleware name 'NewRelicHeader'

### ISSUE THREAD

**Issue by st1906** [aws/aws-sdk-js-v3#4869]
### Checkboxes for prior research

- [X] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest)
- [X] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js).
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js-v3/issues) and didn't find any solution.

### Describe the bug

Hi, with the release of @aws-sdk/ssm-client v3.335.0 https://github.com/aws/aws-sdk-js-v3/releases/tag/v3.335.0, 

started getting the following error on app startup

```
Error: Duplicate middleware name 'NewRelicHeader'
    at Object.add (/***/node_modules/***/node_modules/@aws-sdk/middleware-stack/dist-cjs/MiddlewareStack.js:130:31)
    at Object.applyToStack (/***/node_modules/@newrelic/aws-sdk/lib/v3/smithy-client.js:45:19)
    at Object.use (/***/node_modules/***/node_modules/@aws-sdk/middleware-stack/dist-cjs/MiddlewareStack.js:167:20)
    at SSMClient.wrapShim (/***/node_modules/@newrelic/aws-sdk/lib/v3/util.js:33:26)
    at new WrappedClass (/***/node_modules/newrelic/lib/shim/shim.js:2085:30)
    at new SSMClient (/***/node_modules/***/node_modules/@aws-sdk/client-ssm/dist-cjs/SSMClient.js:27:9)
    at createClient (/***/node_modules/***/dist/services/SSM/createClient.js:5:57)
    at new SSMConnector (/***/node_modules/***/dist/services/SSM/SSMConnector.js:53:55)
    at ***/dist/index.js:34:30
    at Object.<anonymous> (/***/dist/index.js:121:3)
```

### SDK version number

@aws-sdk/ssm-client v3.335.0

### Which JavaScript Runtime is this issue in?

Node.js

### Details of the browser/Node.js/ReactNative version

node 18

### Reproduction Steps

require('newrelic') and use @aws-sdk/ssm-client in the same nodeJS project

### Observed Behavior

On service start, the following error appears: 

```
Error: Duplicate middleware name 'NewRelicHeader'
    at Object.add (/***/node_modules/***/node_modules/@aws-sdk/middleware-stack/dist-cjs/MiddlewareStack.js:130:31)
    at Object.applyToStack (/***/node_modules/@newrelic/aws-sdk/lib/v3/smithy-client.js:45:19)
    at Object.use (/***/node_modules/***/node_modules/@aws-sdk/middleware-stack/dist-cjs/MiddlewareStack.js:167:20)
    at SSMClient.wrapShim (/***/node_modules/@newrelic/aws-sdk/lib/v3/util.js:33:26)
    at new WrappedClass (/***/node_modules/newrelic/lib/shim/shim.js:2085:30)
    at new SSMClient (/***/node_modules/***/node_modules/@aws-sdk/client-ssm/dist-cjs/SSMClient.js:27:9)
    at createClient (/***/node_modules/***/dist/services/SSM/createClient.js:5:57)
    at new SSMConnector (/***/node_modules/***/dist/services/SSM/SSMConnector.js:53:55)
    at ***/dist/index.js:34:30
    at Object.<anonymous> (/***/dist/index.js:121:3)
```

### Expected Behavior

No error.

### Possible Solution

Use of @smithy types and protocol-http package might be conflicting with the newrelic package trying to add the same middleware twice.

**Comment by RanVaknin** [comment]
Hi. @st1906 ,

Unfortunately using New Relic requires us to get a license key so its not possible to test with this 3rd party dependency.

Additionally are you sure you are installing the right client? its `@aws-sdk/client-ssm` and not `@aws-sdk/ssm-client`

Can you try deleting `package.json` and `node_modules` and running npm install again? 

Thanks,
Ran~

**Comment by doublesharp** [comment]
I am getting this error with `@aws-skd/client-s3` version 2.1406.0 as well.

```text

Jun 27 14:50:15 [2023-06-27T21:50:15.198Z] error Unhandled Rejection at: Promise [object Promise] error: Duplicate middleware name 'NewRelicHeader' Error: Duplicate middleware name 'NewRelicHeader' 
Jun 27 14:50:15     at Object.add (/app/node_modules/@aws-sdk/client-s3/node_modules/@aws-sdk/middleware-stack/dist-cjs/MiddlewareStack.js:130:31) 
Jun 27 14:50:15     at Object.applyToStack (/app/node_modules/@newrelic/aws-sdk/lib/v3/smithy-client.js:45:19) 
Jun 27 14:50:15     at Object.use (/app/node_modules/@aws-sdk/client-s3/node_modules/@aws-sdk/middleware-stack/dist-cjs/MiddlewareStack.js:167:20) 
Jun 27 14:50:15     at S3.wrapShim (/app/node_modules/@newrelic/aws-sdk/lib/v3/util.js:33:26) 
Jun 27 14:50:15     at new WrappedClass (/app/node_modules/newrelic/lib/shim/shim.js:2085:30) 
Jun 27 14:50:15     at new S3Client (/app/node_modules/@aws-sdk/client-s3/dist-cjs/S3Client.js:32:9) 
Jun 27 14:50:15     at new S3 (/app/node_modules/@aws-sdk/client-s3/dist-cjs/S3.js:194:1) 
Jun 27 14:50:15     at Object.<anonymous> (/app/controllers/upload.js:13:12) 
Jun 27 14:50:15     at Module._compile (node:internal/modules/cjs/loader:1105:14) 
Jun 27 14:50:15     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10) 
Jun 27 14:50:15     at Module.load (node:internal/modules/cjs/loader:981:32) 
Jun 27 14:50:15     at Function.Module._load (node:internal/modules/cjs/loader:822:12) 
Jun 27 14:50:15     at Function.wrappedLoad [as _load] (/app/node_modules/newrelic/lib/shimmer.js:313:24) 
Jun 27 14:50:15     at Module.require (node:internal/modules/cjs/loader:1005:19) 
Jun 27 14:50:15     at Module.Hook._require.Module.require (/home/app/.nvm/versions/node/v16.16.0/lib/node_modules/pm2/node_modules/require-in-the-middle/index.js:101:39) 
Jun 27 14:50:15     at require (node:internal/modules/cjs/helpers:102:18) 

```

**Comment by st1906** [comment]
Hi @RanVaknin , 
I made a typo, you are correct, it is @aws-sdk/client-ssm.
Tried doing what you suggested, but still getting the same error

**Comment by st1906** [comment]
A small update, the last version that does not throw the mentioned error is `3.303`.
After that one, I could only install `3.306` where the error is thrown

**Comment by RanVaknin** [comment]
Hi @st1906 ,

I have looked in our source code, and there is no middleware named "NewRelicHeader". This is likely due to how this 3rd party dependency is implemented or something to do with your code.

The SDK does not guarantee compatibility with 3rd party tools like New Relic and so we cannot support it. I suggest you open this issue in their [Github repo](https://github.com/newrelic/node-newrelic/issues) and include you actual code so they can better assist you. 

Sorry for the inconvenience.

All the best,
Ran~
